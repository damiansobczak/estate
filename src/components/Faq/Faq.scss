@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";


.faq {
    padding: 90px 0;

    &__items {
        padding-top: 60px;
        flex-wrap: wrap;
    }

    &__title {
        color: $color-text-primary;
        text-align: center;
        text-transform: capitalize;
        @include breakpoint-down(md){
            font-size: $font-500;
        }
    }
}