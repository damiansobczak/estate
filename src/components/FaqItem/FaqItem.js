import React from "react";
import "./FaqItem.scss";

class FaqItem extends React.Component {
  render() {
    return (
      <div className="column is-6-tablet">
        <div className="faqitem">
            <div className="faqitem__count">{this.props.count}.</div>
            <div className="faqitem__content">
                <div className="faqitem__title">{this.props.title}</div>
                <div className="faqitem__desc">{this.props.desc}</div>
            </div>
        </div>
      </div>
    );
  }
}

export default FaqItem;
