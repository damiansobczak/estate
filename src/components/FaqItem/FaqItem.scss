@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.faqitem {
    border-top: 1px solid $color-neutral-200;
    display: flex;
    padding: 40px 40px 0 0;
    margin-bottom: 50px;
    box-sizing: border-box;

    &__count {
        color: $color-accent;
        font-family: poppins, sans-serif;
        font-weight: 700;
        font-style: normal;
        margin-right: 40px;
    }

    &__title {
        font-family: poppins, sans-serif;
        font-weight: 700;
        font-style: normal;
        color: $color-text-primary;
        margin-bottom: 15px;
        line-height: 1.2;
    }

    &__desc {
        color: $color-text-secondary;
        font-family: proxima-nova, sans-serif;
        font-weight: 500;
        font-style: normal;
        line-height: 1.5;
    }
}