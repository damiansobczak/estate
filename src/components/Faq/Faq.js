import React from "react";
import FaqItem from "../FaqItem/FaqItem";
import "./Faq.scss";

class Faq extends React.Component {
  render() {
    return (
      <section className="faq">
        <div className="container">
            <h3 className="faq__title">Często zadawane pytania</h3>
            <div className="faq__items columns">
                <FaqItem count="01" title="Jakich dokumentów potrzebuję?" desc="Maecenas in turpis diam. Sed eget metus at urna mattis vehicula. Duis iaculis ligula a enim vestibulum lacinia."/>
                <FaqItem count="02" title="Jakich dokumentów potrzebuję?" desc="Maecenas in turpis diam. Sed eget metus at urna mattis vehicula. Duis iaculis ligula a enim vestibulum lacinia."/>
                <FaqItem count="03" title="Jakich dokumentów potrzebuję?" desc="Maecenas in turpis diam. Sed eget metus at urna mattis vehicula. Duis iaculis ligula a enim vestibulum lacinia."/>
                <FaqItem count="04" title="Jakich dokumentów potrzebuję?" desc="Maecenas in turpis diam. Sed eget metus at urna mattis vehicula. Duis iaculis ligula a enim vestibulum lacinia."/>
            </div>
        </div>
      </section>
    );
  }
}

export default Faq;
