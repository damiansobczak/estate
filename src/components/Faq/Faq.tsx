import React from "react";
import FaqItem from "../FaqItem/FaqItem";
import { FaqProps } from "./Interfaces/FaqProps";
import "./Faq.scss";

class Faq extends React.Component<FaqProps> {
  render() {
    return (
      <section className="faq">
        <div className="container">
          <h3 className="faq__title">CzÄ™sto zadawane pytania</h3>
          <div className="faq__items columns">
            {this.props.data && this.props.data.map((item: {
              id: Number, title: { rendered: String }, content: { rendered: String }
            }, index: any) => <FaqItem count={`0${index + 1}`} title={item.title.rendered} desc={item.content.rendered.replace(/<p>|<\/p>/g, '')} key={index} />)}
          </div>
        </div>
      </section>
    );
  }
}

export default Faq;
