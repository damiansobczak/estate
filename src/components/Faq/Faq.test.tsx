import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Faq from "./Faq";
import FaqItem from "../FaqItem/FaqItem";

const data = [
    {
        id: 0,
        title: {
            rendered: "To jest Faq 1"
        },
        content: {
            rendered: "<p>Nasze biuro nieruchomości pobiera zadatek w kwocie nieprzekraczającej 2% od całkowitej wartości nieruchomości.</p>"
        }
    },
    {
        id: 1,
        title: {
            rendered: "To jest Faq 2"
        },
        content: {
            rendered: "<p>Nasze biuro nieruchomości pobiera zadatek w kwocie nieprzekraczającej 2% od całkowitej wartości nieruchomości.</p>"
        }
    },
    {
        id: 2,
        title: {
            rendered: "To jest Faq 3"
        },
        content: {
            rendered: "<p>Nasze biuro nieruchomości pobiera zadatek w kwocie nieprzekraczającej 2% od całkowitej wartości nieruchomości.</p>"
        }
    },
]

describe("<Faq />", () => {
    it("renders correctly", () => {
        const tree = renderer.create(<Faq data={data} />).toJSON;
        expect(tree).toMatchSnapshot();
    });

    it("renders item correctly", () => {
        const wrapper = shallow(<Faq data={data} />);
        expect(wrapper.find('FaqItem').length).toEqual(3);
    });

    it("shows faq without html tag", () => {
        const wrapper = shallow(<Faq data={data} />);
        expect(wrapper.find('FaqItem').first().html()).not.toContain('<p>');
    });
});
