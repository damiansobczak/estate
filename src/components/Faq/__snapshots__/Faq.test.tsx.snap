// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Faq /> renders correctly 1`] = `[Function]`;
